tools/cldr-code/src/test/java/org/unicode/cldr/unittest/CheckLdmlDtdReferences.java

```diff
@@ -0,0 +1,342 @@
+package org.unicode.cldr.unittest;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.unicode.cldr.util.CLDRPaths;
+import org.unicode.cldr.util.DtdData;
+import org.unicode.cldr.util.DtdData.Attribute;
+import org.unicode.cldr.util.DtdData.Element;
+import org.unicode.cldr.util.DtdType;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.TreeMultimap;
+import com.ibm.icu.impl.Pair;
+
+public class CheckLdmlDtdReferences {
+
+    static Map<DtdType, DtdData> typeToData = ImmutableMap.of(
+        DtdType.ldml, DtdData.getInstance(DtdType.ldml),
+        DtdType.supplementalData, DtdData.getInstance(DtdType.supplementalData),
+        DtdType.ldmlBCP47, DtdData.getInstance(DtdType.ldmlBCP47)
+        );
+
+    static Multimap<DtdType, String> mismatched = TreeMultimap.create();
+    static Multimap<String, String> unrecognized = LinkedHashMultimap.create();
+
+    static Multimap<DtdType, Element> missingElements = LinkedHashMultimap.create();
+    static Multimap<DtdType, Attribute> missingAttributes = LinkedHashMultimap.create();
+    static Map<DtdType, Multimap<Element, String>> foundElementsToLink = new LinkedHashMap<>();
+
+    static Map<DtdType, Multimap<Element, Element>> childToParent = new LinkedHashMap<>();
+//    static Map<DtdType, Multimap<Element, String>> foundAttributesToLink = new LinkedHashMap<>();
+    static {
+        for (DtdType dtdType : typeToData.keySet()) {
+            final DtdData dtdData = typeToData.get(dtdType);
+            missingElements.putAll(dtdType, dtdData.getElements());
+            missingAttributes.putAll(dtdType, dtdData.getAttributes());
+            foundElementsToLink.put(dtdType, LinkedHashMultimap.create());
+            childToParent.put(dtdType, getChildToParent(dtdData, dtdData.ROOT, LinkedHashMultimap.create()));
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        try (Stream<Path> stream = Files.list(Paths.get(CLDRPaths.BASE_DIRECTORY+"docs","ldml"))) {
+            stream.forEach(x -> {
+                try {
+                    if (!Files.isDirectory(x)) {
```

Comment:
```suggestion
                    if (Files.isReadable(x) && x.endsWith(".md") && x.startsWith("tr35") ) {
```

future proof this a bit